src/config-loader/config-loader.ts

```diff
@@ -52,294 +52,267 @@ export interface Config {
 // Setup function
 export const setup = (app: _express.Application) => {
 	const loadConfig = (data: Config): Bluebird<void> =>
-		sbvrUtils.db
-			.transaction(tx =>
-				Bluebird.try(() => {
-					const authApiTx = sbvrUtils.api.Auth.clone({
-						passthrough: {
-							tx,
-							req: permissions.root,
-						},
-					});
+		sbvrUtils.db.transaction(async tx => {
+			const authApiTx = sbvrUtils.api.Auth.clone({
+				passthrough: {
+					tx,
+					req: permissions.root,
+				},
+			});
 
-					if (data.users != null) {
-						const permissionsCache: {
-							[index: string]: Bluebird<number>;
-						} = {};
-						_.each(data.users, user => {
-							if (user.permissions == null) {
-								return;
-							}
-							_.each(user.permissions, permissionName => {
-								if (permissionsCache[permissionName] != null) {
-									return;
-								}
-								permissionsCache[permissionName] = authApiTx
-									.get({
+			const { users } = data;
+			if (users != null) {
+				const permissionsCache: {
+					[index: string]: Promise<number>;
+				} = {};
+				_.each(users, user => {
+					if (user.permissions == null) {
+						return;
+					}
+					_.each(user.permissions, permissionName => {
+						if (permissionsCache[permissionName] != null) {
+							return;
+						}
+						permissionsCache[permissionName] = (async () => {
```

Comment:
Done (also since this is only done once ever at startup the benefits of avoiding the multiple function creation is limited, but still nice to do especially with deduplicating code)

Reply:
Oh I didn't realize that this was only running once :sweat_smile: 
But as you said the code became so much smaller & readable :+1: